dashboard/src/pages/TreeDetails/Tabs/Tests/IndividualTestsTable.tsx

```diff
@@ -151,57 +174,75 @@ export function IndividualTestsTable({
-        </TableHead>
+        </TH>
       );
     });
     // TODO: remove exhaustive-deps and change memo  (all tables)
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [groupHeaders, sorting]);
 
-  const modelRows = table.getRowModel().rows;
+  const { rows } = useMemo(() => {
+    return table.getRowModel();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [table, sorting]);
+
+  const parentRef = useRef<HTMLDivElement>(null);
+  const virtualizer = useVirtualizer({
+    count: rows.length,
+    // eslint-disable-next-line no-magic-numbers
+    estimateSize: () => 60,
+    getScrollElement: () => parentRef.current,
+    overscan: 5,
+  });
+  const virtualItems = virtualizer.getVirtualItems();
+
+  const { pTop: paddingTop, pBottom: paddingBottom } = useMemo(() => {
+    const pTop = virtualItems?.length > 0 ? virtualItems[0].start || 0 : 0;
+    const pBottom =
+      virtualItems?.length > 0
+        ? virtualizer.getTotalSize() - (virtualItems?.at(-1)?.end || 0)
+        : 0;
+    return { pTop, pBottom };
+  }, [virtualItems, virtualizer]);
+
   const tableRows = useMemo((): JSX.Element[] => {
-    return modelRows?.length
-      ? modelRows.map(row => (
-          <TableRow
-            key={row.id}
-            data-state={row.getIsSelected() && 'selected'}
-            className="cursor-pointer hover:bg-lightBlue"
-          >
-            {row.getVisibleCells().map(cell => (
-              <TableCellWithLink
-                key={cell.id}
-                linkProps={{
-                  to: '/tree/$treeId/test/$testId',
-                  params: {
-                    testId: row.original.id,
-                  },
-                  search: s => s,
-                }}
-              >
-                {flexRender(cell.column.columnDef.cell, cell.getContext())}
-              </TableCellWithLink>
-            ))}
-          </TableRow>
-        ))
-      : [
-          <TableRow key="no-results">
-            <TableCell colSpan={columns.length} className="h-24 text-center">
-              <FormattedMessage id="global.noResults" />
-            </TableCell>
-          </TableRow>,
-        ];
-  }, [modelRows]);
+    return virtualItems.map(virtualRow => {
+      const row = rows[virtualRow.index] as Row<TIndividualTest>;
+      return <TableRowMemoized row={row} key={row.id} />;
+    });
+  }, [rows, virtualItems]);
 
   return (
-    <BaseTable headerComponents={tableHeaders}>
-      <TableBody>{...tableRows}</TableBody>
-    </BaseTable>
+    <div
+      ref={parentRef}
+      style={{ height: '300px', overflow: 'auto', maxWidth: '100%' }}
+    >
+      <div className="rounded-lg border-x border-t border-darkGray bg-white text-sm text-black">
+        <table className="w-full">
+          <DumbTableHeader>{tableHeaders}</DumbTableHeader>
+          <TableBody>
+            {paddingTop > 0 && (
+              <tr>
+                <td style={{ height: paddingTop }}></td>
```

Comment:
ditto about unstable reference

Reply:
you can do dynamic variable with tailwind, just define a an scoped custom css attribute using the `<style>` tag

Reply:
you will probably have to create a class for that and add an unique id